### PR TITLE
Update categorizer to have no empty "th" cells per a11y audit

### PR DESCRIPTION
**Pull request by Myranae**
## Summary:
->Not merging until Perseus has been released<-

Based on an a11y audit completed by Level Access, table headers that are empty should use "td" instead of "th" elements. This PR updates the table in categorizer to follow this a11y best practice. It also updates several related snapshots.

After:
image

Issue: LP-13085

## Test plan:
- Confirm the first empty cell in the header row is a "td" cell instead of a "th" cell
- Confirm the appearance is the same for the before and after
- Confirm all checks pass and all tests pass

### CODE REVIEW

File: packages/perseus/src/widgets/categorizer.jsx [hunk 1]
```diff
@@ -285,6 +285,11 @@ const styles = StyleSheet.create({
         verticalAlign: "middle",
     },
 
+    emptyHeaderCell: {
+        backgroundColor: "white",
```
Comment (nedredmond) [inline packages/perseus/src/widgets/categorizer.jsx:289]: question: Why is this necessary? Do we have a global style giving all `td`s a weird background color?

suggestion: Can you use something like `inherit` instead, in case someone tries to style a table and suddenly this cell stands out?

**Review by nedredmond — COMMENTED** [review]
I have a question about styling before I approve!

File: .changeset/weak-zebras-marry.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (handeyeco) [inline .changeset/weak-zebras-marry.md:1]: Since this is a user-facing change and not just dev facing, I think it should maybe be a patch?

File: packages/perseus/src/widgets/categorizer.jsx (hunk 1 above)
Comment (nedredmond) [reply packages/perseus/src/widgets/categorizer.jsx:289]: Or `unset` or `currentColor` even maybe `transparent`... any number of others that don't hardcode a background color but default to whatever the test of the DOM is into

**Review by nedredmond — APPROVED** [review]
Thanks for updating the background color and changeset! LGTM 🚢
